content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -807,6 +807,12 @@ These are some requirements and semantics of the API:
   - `Count`: use to indicate that the Pod should be handled in the default way.
      The counter towards the `.spec.backoffLimit` should be incremented.
 
+{{< note >}}
+Since the 1.27 version Kubelet transitions deleted pods to a terminal phase
+(see: [Pod Phase](/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase)). This
+ensures that deleted pods have their finalizers removed by the Job controller.
```

Comment:
```
When you use a `podFailurePolicy`, the job controller only matches Pods in the `Failed` phase.
Pods with a deletion timestamp that are not in a terminal phase (`Failed` or `Completed`) are considered still terminating.
This implies that terminating pods retain a [tracking finalizer](#job-tracking-with-finalizers) until
they reach a terminal phase.
Since Kubernetes v1.27, Kubelet....
```